src/expr/src/expr/expr_binary_nonnull.rs

```diff
@@ -415,6 +417,38 @@ fn build_at_time_zone_expr(
     Ok(expr)
 }
 
+fn build_cast_with_time_zone_expr(
+    ret: DataType,
+    l: BoxedExpression,
+    r: BoxedExpression,
+) -> Result<BoxedExpression> {
```

Comment:
Are you also going to implement the (implicit) cast between `timestamp with timezone` and `timestamp`? 

As I mentioned in the Slack channel, an alternative choice is to fixate on the UTC timezone to avoid that conversion. Anyway, I am just trying to prevent it from becoming too complex/cumbersome, so I will definitely +1 if you think that's not too complex.

Reply:
Hmmm. It was mentioned in the PR description, but I didn't see it in code. 👀

Reply:
It would rewrite cast expr between them to `AT TIME ZONE`

Reply:
Yes, it's covered here.